homeassistant/helpers/deprecation.py

```diff
@@ -164,6 +165,30 @@ def _print_deprecation_warning_internal(
     breaks_in_ha_version: str | None,
     *,
     log_when_no_integration_is_found: bool,
+) -> None:
+    # Suppress ImportError due to use of deprecated enum in core.py
+    # Can be removed in HA Core 2025.1
+    with suppress(ImportError):
+        _print_deprecation_warning_internal_impl(
+            obj_name,
+            module_name,
+            replacement,
+            description,
+            verb,
+            breaks_in_ha_version,
+            log_when_no_integration_is_found=log_when_no_integration_is_found,
+        )
```

Comment:
The issue is these deprecated constants:
https://github.com/home-assistant/core/blob/335104dcc8c9afc30e6641d2a2a37fb34bf968fa/homeassistant/core.py#L179-L185

Since they will be removed in 2 months, I did not want to make a more complicated solution here